src/Api/Upload.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Kirby\Api;
+
+use Kirby\Cms\App;
+use Kirby\Filesystem\Dir;
+use Kirby\Filesystem\F;
+use Kirby\Toolkit\Str;
+
+/**
+ * The Upload class supports file uploads in the
+ * context of the API
+ *
+ * @package   Kirby Api
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     4.3.0
+ */
+class Upload
+{
+	/**
+	 * Handle chunked uploads by merging all chunks
+	 * in the tmp directory and only returning the new
+	 * $source path to the tmp file once complete
+	 */
+	public static function chunk(
+		Api $api,
+		string $source,
+		string $filename
+	): string|null {
+		// if the file is uploaded in chunks
+		if ($isChunk = $api->requestHeaders('X-Upload-Chunk')) {
+			// ensure the directory exists
+			Dir::make($dir = static::tmp());
+
+			// append chunk to tmp file with unique filename (id suffix)
+			$id     = $api->requestHeaders('X-Unique-Upload-Id');
```

Comment:
Needs to be filtered for safe characters, otherwise we would have a major directory traversal attack vector.

Reply:
`Str::slug($id, '', 'a-z')`?